app/js/features/license/saga.js

```diff
@@ -32,29 +35,61 @@ export function *doCheckLicenseOwner() {
     const isLicenseOwner = yield call(License.methods.isLicenseOwner(web3.eth.defaultAccount).call);
     yield put({type: CHECK_LICENSE_OWNER_SUCCEEDED, isLicenseOwner});
   } catch (error) {
-    yield put({type: CHECK_LICENSE_OWNER_FAILED, error});
+    console.error(error);
+    yield put({type: CHECK_LICENSE_OWNER_FAILED, error: error.message});
   }
 }
 
 export function *onCheckLicenseOwner() {
   yield takeEvery(CHECK_LICENSE_OWNER, doCheckLicenseOwner);
 }
 
-export async function *doUserRating() {
+export function *doGetLicenseOwners() {
+  try {
+    // TODO get more information like position and rate
+    const events = yield License.getPastEvents('Bought', {fromBlock: 1});
+    const licenseOwners = events.map(event => {
+      return {address: event.returnValues.buyer};
+    });
+    yield put({type: GET_LICENSE_OWNERS_SUCCCEDED, licenseOwners});
+  } catch (error) {
+    console.error(error);
+    yield put({type: GET_LICENSE_OWNERS_FAILED, error: error.message});
+  }
+}
+
+export function *onGetLicenseOwners() {
+  yield takeEvery(GET_LICENSE_OWNERS, doGetLicenseOwners);
+}
+
+export function *checkUserRating() {
   try {
     const arrAvg = arr => arr.reduce((a,b) => a + b, 0) / arr.length;
-    const events = await Escrow.getPastEvents('Rating', {fromBlock: 1, filter: {seller: web3.eth.defaultAccount}});
+    const events = yield Escrow.getPastEvents('Rating', {fromBlock: 1, filter: {seller: web3.eth.defaultAccount}});
     const ratings = events.slice(events.length - 5).map((e) => parseInt(e.returnValues.rating, 10));
     const averageRating = arrAvg(ratings);
 
     yield put({type: USER_RATING_SUCCEEDED, userRating: averageRating});
   } catch (error) {
-    yield put({type: USER_RATING_FAILED, error});
+    console.error(error);
+    yield put({type: USER_RATING_FAILED, error: error.message});
   }
 }
-
 export function *onUserRating() {
-  yield takeEvery(USER_RATING, doUserRating);
+  yield takeEvery(USER_RATING, checkUserRating);
+}
+
+export async function *addRating() {
+  try {
+    //rate_transaction(uint _escrowId, uint _rate)
```

Comment:
Is it on purpose?

Reply:
Yes and no. Ignore everything "rate" related